fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2707,25 +2707,27 @@ public void addRebuildRecordCountsJob(List<CompletableFuture<Void>> work) {
     }
 
     /**
-     * Return a list of boundaries separating the key ranges maintained by each FDB server. This information can be
+     * Return a cursor of boundaries separating the key ranges maintained by each FDB server. This information can be
      * useful for splitting a large task (e.g., rebuilding an index for a large record store) into smaller tasks (e.g.,
      * rebuilding the index for records in certain primary key ranges) more evenly so that they can be executed in a
      * parallel fashion efficiently. The returned boundaries are an estimate from FDB's locality API and may not
      * represent the exact boundary locations at any database version.
      * <p>
-     * The boundaries are returned as a list which is sorted and does not contain any duplicates. The first element of
+     * The boundaries are returned as a cursor which is sorted and does not contain any duplicates. The first element of
      * the list is greater than or equal to <code>low</code>, and the last element is less than or equal to
      * <code>high</code>.
      * <p>
      * This implementation may not work when there are too many shard boundaries to complete in a single transaction.
+     * <p>
+     * This cursor is blocking because an async iterator is passed in {@link IteratorCursor}.c
```

Comment:
The overall change is fine, but there's a trailing `c` after the period.  Also, I think you should leave the "because an async iterator is passed in..." off because for someone just reading the java docs it is terribly confusing because there is no `IteratorCursor` that is passed in.  I think you can just leave it as something like: "Note: the returned cursor is blocking and must not be used in an asynchronous context".

Reply:
Fixed.